quaddtype/numpy_quaddtype/src/ops.hpp

```diff
@@ -1008,6 +1008,43 @@ quad_nextafter(const Sleef_quad *x, const Sleef_quad *y)
     return quad_set_words64(hx, lx);
 }
 
+static inline Sleef_quad
+quad_spacing(const Sleef_quad *x)
+{   
+    // spacing(x) returns the distance between x and the next representable value
+    // The result has the SAME SIGN as x (NumPy convention)
+    // For x >= 0: spacing(x) = nextafter(x, +inf) - x
+    // For x < 0:  spacing(x) = nextafter(x, -inf) - x (negative result)
+    
+    // Handle NaN
+    if (Sleef_iunordq1(*x, *x)) {
+        return *x;  // NaN
+    }
+    
+    // Handle infinity -> NaN (numpy convention)
+    if (quad_isinf(x)) {
+        return QUAD_NAN;
+    }
+    
+    // Determine direction based on sign of x
+    Sleef_quad direction;
+    if (Sleef_icmpltq1(*x, QUAD_ZERO)) {
+        // Negative: move toward -inf
+        direction = Sleef_negq1(QUAD_POS_INF);
+    } else {
+        // Positive or zero: move toward +inf
```

Comment:
Just checked and the spacing for -0.0 and +0.0 is the same ... sigh

Reply:
yeah same as curent numpy
```
In [2]: np.spacing(-0.0)
Out[2]: np.float64(5e-324)

In [3]: np.spacing(0.0)
Out[3]: np.float64(5e-324)
```